client/signup/steps/import-url/index.jsx

```diff
@@ -15,34 +15,43 @@ import Button from 'components/button';
 import ButtonGroup from 'components/button-group';
 import StepWrapper from 'signup/step-wrapper';
 import SignupActions from 'lib/signup/actions';
-import FormTextInputWithAction from 'components/forms/form-text-input-with-action';
-//import FormInputValidation from 'components/forms/form-input-validation';
-import { fetchIsSiteImportable, setNuxUrlInputValue } from 'state/importer-nux/actions';
+import FormTextInput from 'components/forms/form-text-input';
+import FormInputValidation from 'components/forms/form-input-validation';
+import { setNuxUrlInputValue, setValidationMessage } from 'state/importer-nux/actions';
 import {
 	getNuxUrlInputValue,
 	getSiteDetails,
+	getUrlInputValidationMessage,
 	isUrlInputDisabled,
 } from 'state/importer-nux/temp-selectors';
 
 const debug = debugFactory( 'calypso:signup-step-import-url' );
 
-const DEBOUNCE_CHECK_INTERVAL = 1200;
+const VALIDATION_INTERVAL = 1200;
 
 const normalizeUrlForImportSource = url => {
 	// @TODO sanitize? Prepend https:// ..?
 	return url;
 };
 
+const isValidUrl = ( value = '' ) =>
```

Comment:
This one is too simple to validate most common cases we need. I would prefer we switch from RegExp to some lib. I've seen that we have [`valid-url`](https://www.npmjs.com/package/valid-url) installed so I would go for that one.

Things I've tested and haven't worked:
- https://marekhrabe.com - when you copy an URL from Chrome, the protocol is always present. We definitely need to allow that as a primary way and maybe later accomodate for protocol-less ones
- marekhrabe.wixsite.com/adi-test123 - this one is very important - the free site address currently doesn't work

Reply:
I tried `valid-url` but it seemed _too_ strict in that any non-protocol'ed url was rejected. I think it'd be unreasonable to expect everyone in the signup flow to type `http://`/`https://` correctly - some might not understand which one to use etc.

To avoid being too strict with the protocol, I've tried to use `valid-url` but prepending `http://` in cases where no protocol was found, something like this:

```js
const { protocol } = parseURL( value );
return protocol
  ? isWebUri( value )
  : isWebUri( 'http://' + value );
```

This still allows some values through that would cause issues for us, like: `http://yo` so I've added a test for a `.` in the `hostname`.

This seems to be working somewhat... commit coming up :)